src/rules.js

```diff
@@ -318,8 +318,8 @@ inline.gfm = merge({}, inline.normal, {
   _extended_email: /[A-Za-z0-9._+-]+(@)[a-zA-Z0-9-_]+(?:\.[a-zA-Z0-9-_]*[a-zA-Z0-9])+(?![-_])/,
   url: /^((?:ftp|https?):\/\/|www\.)(?:[a-zA-Z0-9\-]+\.?)+[^\s<]*|^email/,
   _backpedal: /(?:[^?!.,:;*_~()&]+|\([^)]*\)|&(?![a-zA-Z0-9]+;$)|[?!.,:;*_~)]+(?!$))+/,
-  del: /^~+(?=\S)([\s\S]*?\S)~+/,
-  text: /^(`+|[^`])(?:(?= {2,}\n)|[\s\S]*?(?:(?=[\\<!\[`*~]|\b_|https?:\/\/|ftp:\/\/|www\.|$)|[^ ](?= {2,}\n)|[^a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-](?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@))|(?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@))/
+  del: /^(~~?)(?=[^\s~])([\s\S]*?[^\s~])\1(?=[^~]|$)/,
+  text: /^([`~]+|[^`~])(?:(?= {2,}\n)|[\s\S]*?(?:(?=[\\<!\[`*~]|\b_|https?:\/\/|ftp:\/\/|www\.|$)|[^ ](?= {2,}\n)|[^a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-](?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@))|(?=[a-zA-Z0-9.!#$%&'*+\/=?_`{\|}~-]+@))/
```

Comment:
@davisjam I don't think these are any more susceptible to ReDOS than the old regexs but could you check them out?

Here are the railroad diagrams: [del](https://uzitech.github.io/regexper/#%7Cflavor%3Djavascript%7C%2F%5E%28~~%3F%29%28%3F%3D%5B%5E%5Cs~%5D%29%28%5B%5Cs%5CS%5D*%3F%5B%5E%5Cs~%5D%29%5C1%28%3F%3D%5B%5E~%5D%7C%24%29%2F) [text](https://uzitech.github.io/regexper/#%7Cflavor%3Djavascriptes5%7C%2F%5E%28%5B%60~%5D%2B%7C%5B%5E%60~%5D%29%28%3F%3A%28%3F%3D%20%7B2%2C%7D%5Cn%29%7C%5B%5Cs%5CS%5D*%3F%28%3F%3A%28%3F%3D%5B%5C%5C%3C!%5C%5B%60*~%5D%7C%5Cb_%7Chttps%3F%3A%5C%2F%5C%2F%7Cftp%3A%5C%2F%5C%2F%7Cwww%5C.%7C%24%29%7C%5B%5E%20%5D%28%3F%3D%20%7B2%2C%7D%5Cn%29%7C%5B%5Ea-zA-Z0-9.!%23%24%25%26'*%2B%5C%2F%3D%3F_%60%7B%5C%7C%7D~-%5D%28%3F%3D%5Ba-zA-Z0-9.!%23%24%25%26'*%2B%5C%2F%3D%3F_%60%7B%5C%7C%7D~-%5D%2B%40%29%29%7C%28%3F%3D%5Ba-zA-Z0-9.!%23%24%25%26'*%2B%5C%2F%3D%3F_%60%7B%5C%7C%7D~-%5D%2B%40%29%29%2F)

Reply:
[safe-regex](https://runkit.com/uzitech/5fb085d1a53c4700254cb346) says they are safe.